src/app/courses/courses_store/courses.effects.ts

```diff
@@ -0,0 +1,80 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Router } from "@angular/router";
+import { Actions, createEffect, ofType } from "@ngrx/effects";
+import { Store } from "@ngrx/store";
+import { map, switchMap, tap, withLatestFrom } from "rxjs";
+
+import * as CoursesActions from './course.actions'
+import { HttpCourse } from "src/app/shared/interfaces/http-course.interface";
+import * as fromApp from '../../store/app.reducer'
+
+export interface CourseResponseData {
+    courses: HttpCourse[]
+}
+
+
+@Injectable()
+export class CourseEffects {
+    private apiUrl = 'http://localhost:3004';
+
+    setCourses$ = createEffect(() => 
+        this.actions$.pipe(
+            ofType(CoursesActions.FetchCourses),
+            withLatestFrom(this.store.select('courses')),
+            switchMap(([actionData, state]) => {
+                return this.http.get<HttpCourse[]>(
+                    `${this.apiUrl}/courses?start=0&count=${state.quantityOfVisibleCourses}`
```

Comment:
_quantityOfVisibleCourses_ not the best naming
Maybe change it to _shownCoursesQuantity_

Reply:
ok, agree, changed